### PR TITLE
Primary CI pipeline specified in yml

### PR DESCRIPTION
**Pull request by grigoryk**
Transitioning away from the classic ADO pipeline for our main CI to a YAML one.

### CODE REVIEW

**Comment by grigoryk** [comment]
Please don't merge yet, still verifying it works well but want to see how it shows up in a github PR.

**Comment by grigoryk** [comment]
Okay, I'm pretty happy with this. It's a one-to-one merge of the existing CI pipeline, except quite a bit streamlined (exported YAML is much longer/messier than hand-written one). Also, I've adjusted branch/PR triggers so that the pipeline doesn't automatically run on all branch pushes, like the legacy one did; that seems excessive. With the new definition, it'll run on PRs against main/develop/release*, and on commits/merges to these integration branches. All other runs can be done manually, if required.

**Comment by grigoryk** [comment]
Next steps after merging:
- specify the new pipeline in our ServiceTree metadata
- remove the old one, `SCC-SPOOL-communication-ui-library-android CI`
- rename the new pipeline in ADO (remove the 'yaml' annotation)
- rebase and land https://github.com/Azure/communication-ui-library-android/pull/620 to get broader test coverage reporting

File: azure-pipelines-ci.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+pool:
+  vmImage: 'macos-latest'
+
+steps:
+- task: servicetree-link-build-task@1
+  displayName: 'ServiceTree Integration'
+  inputs:
+    ServiceTreeGateway: 'ServiceTree Gateway'
+    Service: 'e312ec77-9c77-48a7-9837-32fe6d08d460'
+    BuildOutputUsage: 'test'
```
Comment (iaulakh) [inline azure-pipelines-ci.yml:27]: is it safe to share this information as public? is it possible to use secret inside azure devops to get these values?

**Review by iaulakh — DISMISSED** [review]

File: azure-pipelines-ci.yml (hunk 1 above)
Comment (grigoryk) [reply azure-pipelines-ci.yml:27]: Yup, something we're already doing in other Azure repositories.

**Review by ahammer — DISMISSED** [review]

**Review by iaulakh — DISMISSED** [review]

**Review by ahammer — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Pull request merged**